src/core/lwt.mli

```diff
@@ -1377,10 +1377,14 @@ let () =
       This operator is obscure and its use is discouraged. It is the same as
       [p >|= f]. *)
 
-  (** This module provides support for {{:https://github.com/janestreet/ppx_let}
-      ppx_let}.
+end
+
+(** This module provides support for {{:https://github.com/janestreet/ppx_let}
+    ppx_let}.
 
-      @since 4.2.0 *)
+    @since 4.2.0 *)
+module Let_syntax :
```

Comment:
Could you avoid a breaking change by just creating a new module `Let_syntax.Let_syntax`, without *moving* the existing `Syntax.Let_syntax`? That is, keep `Syntax.Let_syntax` around.

If it has to be a breaking change, I recommend splitting this PR into the `Lwt_result` part and the breaking part, because the breaking change will likely take much longer to merge (for a major release).

Reply:
> Could you avoid a breaking change by just creating a new module Let_syntax.Let_syntax, without moving the existing Syntax.Let_syntax? That is, keep Syntax.Let_syntax around.

That's a good point. I've updated the PR keeping the old `Let_syntax` module in place.

Reply:
I'll need to read up on how to achive this without causing an error on ocaml < 4.08

```
Error: Multiple definition of the module name Let_syntax.

       Names must be unique in a given structure or signature.
```

Reply:
Looks like my newest commit fixes the build error on older compiler versions